src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataViewModelSorter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Windows.Data;
+using System.Windows.Markup;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Sorts immutable collections of <see cref="DocumentSymbolDataViewModel"/>s 
+    /// </summary>
+    internal class DocumentSymbolDataViewModelSorter : MarkupExtension, IMultiValueConverter
+    {
+        public static DocumentSymbolDataViewModelSorter Instance { get; } = new();
```

Comment:
Does this need (or should this have) a private constructor to avoid creation elsewhere?